lib/codegen/src/write.rs

```diff
@@ -9,9 +9,36 @@ use packed_option::ReservedValue;
 use std::fmt::{self, Write};
 use std::string::String;
 
+pub fn write_function_plain(
+    w: &mut Write,
+    func: &Function,
+    isa: Option<&TargetIsa>,
+    inst: Inst,
+    indent: usize,
+) -> fmt::Result {
+    write_instruction(w, func, isa, inst, indent)?;
+    Ok(())
+}
+
 /// Write `func` to `w` as equivalent text.
 /// Use `isa` to emit ISA-dependent annotations.
 pub fn write_function(w: &mut Write, func: &Function, isa: Option<&TargetIsa>) -> fmt::Result {
+    decorate_function(
+        &mut |w, func, isa, inst, indent| write_function_plain(w, func, isa, inst, indent),
+        w,
+        func,
+        isa,
+    )
+}
+
+pub fn decorate_function<
```

Comment:
It'd be good to have a brief documentation comment here describing what `decorate_function` does.